components/base/kinematicbase/ptgKinematics.go

```diff
@@ -240,13 +248,15 @@ func (ptgk *ptgBaseKinematics) goToInputs(ctx context.Context, inputs []referenc
 				ptgk.inputLock.Unlock()
 				utils.SelectContextOrWait(ctx, time.Duration(inputUpdateStep*1000*1000)*time.Microsecond)
 			}
+			wg.Done()
```

Comment:
I think this should be deferred at the top of the function as well

Reply:
deferring wg.Wait causes for a data race in `gotoinputs`

Reply:
Something about these data races seems wrong to me since what Peter and I are proposing are Golang idioms.  I'm also not sure why this file would even need to change

Reply:
The data race issue is at least in part because the above `for` loop does not break on context cancelled, and will always execute to completion.

Reply:
> The data race issue is at least in part because the above for loop does not break on context cancelled, and will always execute to completion.

I think that is just a part of the existing data race.
Suppose we would like to run some `SetVelocity` call on a base for exactly 3.13 seconds.
We then enter 
```
utils.PanicCapturingGo(func() {
	// We need to update currentInputs as we move through the arc.
	// note inputUpdateStep == 0.1 seconds
	for timeElapsed := 0.; timeElapsed <= step.timestepSeconds; timeElapsed += inputUpdateStep {
		if ctx.Err() != nil {
			ptgk.logger.CDebug(ctx, ctx.Err().Error())
			// context cancelled
			break
		}
		distIncVel := step.linVelMMps.Y
		if distIncVel == 0 {
			distIncVel = step.angVelDegps.Z
		}
		ptgk.inputLock.Lock()
		ptgk.currentInput = []referenceframe.Input{inputs[0], inputs[1], {math.Abs(distIncVel) * timeElapsed}}
		ptgk.inputLock.Unlock()
		utils.SelectContextOrWait(ctx, time.Duration(inputUpdateStep*1000*1000)*time.Microsecond)
	}
})
```
since this is spawns a go-routine we proceed to 
```
if !utils.SelectContextOrWait(ctx, timestep) {
	ptgk.logger.CDebug(ctx, ctx.Err().Error())
	// context cancelled
	break
}
```
---------------------

Now, `!utils.SelectContextOrWait` will take `3.13` seconds, but the for loop within `PanicCapturingGo` will take slightly longer. The reason why it will take slightly longer is because we will be doing `3.13 / 0.1 = 31.3 => 32` iterations. Since each iteration lasts `0.1` seconds `PanicCapturingGo` will take exactly 3.2 seconds.

Hence, we need to explicitly specify `wg.Wait()` instead of deferring it.